src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8368,18 +8428,38 @@ private BoundExpression BindMemberOfType(
             return result;
         }
 
-        protected MethodGroupResolution BindExtensionMethod(
+        private BoundExpression BindTypeMemberOfType(BoundExpression left, string plainName, NamedTypeSymbol symbol,
+            SyntaxNode node, SyntaxNode right, BindingDiagnosticBag diagnostics, ref bool wasError)
+        {
+            if (IsInstanceReceiver(left) == true && !wasError)
+            {
+                // CS0572: 'B': cannot reference a type through an expression; try 'A.B' instead
+                Error(diagnostics, ErrorCode.ERR_BadTypeReference, right, plainName, symbol);
+                wasError = true;
+            }
+
+            return new BoundTypeExpression(
+                syntax: node,
+                aliasOpt: null,
+                boundContainingTypeOpt: left as BoundTypeExpression,
+                boundDimensionsOpt: ImmutableArray<BoundExpression>.Empty,
+                typeWithAnnotations: TypeWithAnnotations.Create(symbol));
+        }
+
+#nullable enable
+        protected MethodGroupResolution ResolveExtension(
             SyntaxNode expression,
-            string methodName,
-            AnalyzedArguments analyzedArguments,
+            string memberName,
+            AnalyzedArguments? analyzedArguments,
             BoundExpression left,
             ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations,
             OverloadResolution.Options options,
             RefKind returnRefKind,
-            TypeSymbol returnType,
+            TypeSymbol? returnType,
             bool withDependencies,
```

Comment:
>bool withDependencies, [](http://example.com/codeflow?start=12&length=22)

Here, I think that instead of passing ```withDependencies```, we should pass ```ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo```. And keep passing it down and collecting use-site diagnostics into it when lookup might be failing because of them, and in other similar cases.